core-rust/state-manager/src/jni/node_rust_environment.rs

```diff
@@ -199,3 +219,19 @@ impl JNINodeRustEnvironment {
 }
 
 pub fn export_extern_functions() {}
+
+/// Starts a background thread responsible for periodic raw DB metrics collection, and returns a
+/// handle that keeps it running.
+/// See [`RAW_DB_MEASUREMENT_INTERVAL`] for more details.
+fn start_raw_db_metrics_reporting(
+    database: Arc<RwLock<StateManagerDatabase>>,
+    metric_registry: &Registry,
+) -> ScheduleHandle {
+    let raw_db_metrics = RawDbMetrics::new(metric_registry);
+    let mut scheduler = Scheduler::new();
+    scheduler.every(RAW_DB_MEASUREMENT_INTERVAL).run(move || {
+        let statistics = database.read().get_data_volume_statistics();
+        raw_db_metrics.update(statistics);
+    });
+    scheduler.watch_thread(Duration::from_secs(1))
```

Comment:
What does this do out of interest?

Reply:
it literally does `thread::spawn()`, and returns a super-nice handle that will stop the scheduler when dropped.
I would say that conceptually it finalizes the build of the `{scheduler + its tasks}` (since it consumes `self`).

Reply:
And is the `Duration` is a "graceful delay before I kill you on stop command" thing?

Reply:
no, it is a lame "how often the scheduler thread wakes up to check whether any task run is due" a.k.a. "precision"
(which should not be needed at all in a non-lame scheduler implementation)